backend/danswer/db/document_set.py

```diff
@@ -0,0 +1,282 @@
+from collections.abc import Sequence
+from typing import cast
+from uuid import UUID
+
+from sqlalchemy import and_
+from sqlalchemy import delete
+from sqlalchemy import func
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from danswer.db.models import ConnectorCredentialPair
+from danswer.db.models import Document
+from danswer.db.models import DocumentByConnectorCredentialPair
+from danswer.db.models import DocumentSet as DocumentSetDBModel
+from danswer.db.models import DocumentSet__ConnectorCredentialPair
+from danswer.server.models import DocumentSetCreationRequest
+from danswer.server.models import DocumentSetUpdateRequest
+
+
+def _delete_document_set_cc_pairs(
```

Comment:
Can we come up with a name for this? I called it something something no commit at one point, lets have it in the function so there's not a bug that crops up later

Reply:
Let's do `__no_comit`